src/libraries/System.Net.Quic/tests/FunctionalTests/MsQuicRemoteExecutorTests.cs

```diff
@@ -20,27 +20,43 @@ public class MsQuicRemoteExecutorTests : QuicTestBase
         public MsQuicRemoteExecutorTests()
             : base(null!) { }
 
-        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
-        public void SslKeyLogFile_IsCreatedAndFilled()
+        [ConditionalTheory(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void SslKeyLogFile_IsCreatedAndFilled(bool enabledBySwitch)
         {
-            if (PlatformDetection.IsReleaseLibrary(typeof(QuicConnection).Assembly))
+            if (PlatformDetection.IsDebugLibrary(typeof(QuicConnection).Assembly) && !enabledBySwitch)
             {
-                throw new SkipTestException("Retrieving SSL secrets is not supported in Release mode.");
+                throw new SkipTestException("AppCtxSwitch is not necessary for SSLKEYLOGFILE in Debug.");
```

Comment:
But it doesn't hurt, no? It will still work and log the secrets, no? I'm trying to understand why we should skip this combo.

Reply:
For this combination, the test asserts that we *don't* log the secrets if the appctx switch is not present, but in debug builds we log them anyway. Since there is the second test case (when appctx switch is set) which already exercises that path, I thought it better to just skip this test case.

Reply:
What I was getting at is there's no reason to skip this combo as it might be used like that. It's not "forbidden" combination. But if you want to skip it, I'd suggest direct `return` instead of the `Skip` exception as that will spam the output.

Reply:
I skipped it because the asserts at the and have different expectations and the test would fail on Debug builds. The return instead of skip is a good idea.